### PR TITLE
feat: verifying and enabling necessary extensions for ui

### PR DESCRIPTION
**Pull request by aokirisaki**
**What type of PR is this?**

**Which issue does this PR fix**:
#1295 

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -29,10 +29,18 @@ OS ?= linux
 ARCH ?= amd64
 BENCH_OUTPUT ?= stdout
 EXTENSIONS ?= sync,search,scrub,metrics,lint,ui,mgmt
+UI_DEPENDENCIES := search,mgmt
 comma:= ,
+space := $(null) #
```
Comment (andaaron) [inline Makefile:34]:
```suggestion
space := $(null)
```

File: pkg/cli/root.go [hunk 2]
```diff
@@ -312,6 +312,20 @@ func validateCacheConfig(cfg *config.Config) error {
 }
 
 func validateExtensionsConfig(cfg *config.Config) error {
+	if cfg.Extensions != nil && cfg.Extensions.UI != nil && cfg.Extensions.UI.Enable != nil && *cfg.Extensions.UI.Enable {
+		if cfg.Extensions.Mgmt == nil || !*cfg.Extensions.Mgmt.Enable {
+			log.Warn().Err(errors.ErrBadConfig).Msg("Mgmt extension is disabled")
```
Comment (andaaron) [inline pkg/cli/root.go:317]: You should also mention why mgmt being disabled is not acceptable in this case.
Same for search.

File: Makefile [hunk 3]
```diff
@@ -31,10 +31,18 @@ OS ?= linux
...
 hyphen:= -
 extended-name:=
 
+define unique =
```
Comment (andaaron) [inline Makefile:40]: it seems to me like half the logic is in the `BUILD_LABELS` assignment statement and half in this function, and the reasoning for splitting it this way is not clear. `unique` seems to do a very specific thing and cannot be called for another purpose somewhere else. Maybe a name change for it should be done as well, after the logic is refactored.

File: Makefile [hunk 4]
```diff
@@ -31,11 +31,19 @@ TESTDATA := $(TOP_LEVEL)/test/data
 OS ?= linux
 ARCH ?= amd64
 BENCH_OUTPUT ?= stdout
-EXTENSIONS ?= sync,search,scrub,metrics,lint,ui,mgmt,userprefs
+EXTENSIONS ?= sync,search,scrub,metrics,lint,ui,mgmt
+UI_DEPENDENCIES := search,mgmt
```
Comment (andaaron) [inline Makefile:35]:
```suggestion
UI_DEPENDENCIES := search,mgmt,userprefs
```

File: pkg/cli/root.go [hunk 5]
```diff
@@ -314,6 +314,20 @@ func validateCacheConfig(cfg *config.Config) error {
...
 
 func validateExtensionsConfig(cfg *config.Config) error {
+	if cfg.Extensions != nil && cfg.Extensions.UI != nil && cfg.Extensions.UI.Enable != nil && *cfg.Extensions.UI.Enable {
+		if cfg.Extensions.Mgmt == nil || !*cfg.Extensions.Mgmt.Enable {
+			log.Warn().Err(errors.ErrBadConfig).Msg("UI functionality can't be used without mgmt extension.")
+
+			return errors.ErrBadConfig
+		}
+
+		if cfg.Extensions.Search == nil || !*cfg.Extensions.Search.Enable {
```
Comment (andaaron) [inline pkg/cli/root.go:324]: Will need a similar checks for the `userprefs` extension.

File: examples/config-ui.json [hunk 6]
```diff
@@ -18,6 +18,9 @@
     },
     "ui": {
       "enable": true
+    },
+    "mgmt": {
+      "enable": true
```
Comment (andaaron) [inline examples/config-ui.json:23]: Add userprefs as well.

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**